src/config.ts

```diff
@@ -18,34 +19,15 @@ const dataPath = envVar("DATA_PATH")
 ensureDirSync(dataPath)
 
 const appDbVersionPath = path.join(dataPath, "task-db-version")
-const shouldClearTaskDatabaseOnStart = process.env.TASK_DB_VERSION
-  ? ((appDbVersion) => {
-      const currentDbVersion = (() => {
-        try {
-          return readFileSync(appDbVersionPath).toString().trim()
-        } catch (error) {
-          if (
-            /*
-          Test for the following error:
-            [Error: ENOENT: no such file or directory, open '/foo'] {
-              errno: -2,
-              code: 'ENOENT',
-              syscall: 'unlink',
-              path: '/foo'
-            }
-          */
-            !isError(error) ||
-            (error as { code?: string })?.code !== "ENOENT"
-          ) {
-            throw error
-          }
-        }
-      })()
-      if (currentDbVersion !== appDbVersion) {
-        writeFileSync(appDbVersionPath, appDbVersion)
+const taskDbVersion = process.env.TASK_DB_VERSION?.trim() || ""
+const currentDbVersion = getCurrentDbVersion(appDbVersionPath)
+const shouldClearTaskDatabaseOnStart = taskDbVersion
+  ? (() => {
+      if (currentDbVersion !== taskDbVersion) {
+        writeFileSync(appDbVersionPath, taskDbVersion)
         return true
       }
-    })(process.env.TASK_DB_VERSION.trim())
+    })()
```

Comment:
Feels like it could be simplified like this
```ts
const shouldClearTaskDatabaseOnStart = taskDbVersion && currentDbVersion !== taskDbVersion;
if (shouldClearTaskDatabaseOnStart) {
    writeFileSync(appDbVersionPath, taskDbVersion)
}
```

Reply:
the only thing is that if writeFile will fail, the shouldClearTask could still be `true`, not sure if that would be an expected fall back. I'd not clear probably then. So I moved it into named function in db module